### PR TITLE
Add tooltip for fields overridden by Advanced Sync Rules

### PR DESCRIPTION
**Pull request by praveen-elastic**
## Closes https://github.com/elastic/connectors-py/issues/1015, #1016  

The tooltip on the configurable fields informs the user that the field will be ignored when Advanced Sync Rules are used.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/confluence.py [hunk 1]
```diff
@@ -262,6 +262,7 @@ def get_default_configuration(cls):
             "spaces": {
                 "display": "textarea",
                 "label": "Confluence space keys",
+                "tooltip": "This configurable field is ignored when Advanced Sync Rules are used.",
```
Comment (navarone-feekery) [inline connectors/sources/confluence.py:265]: Could you keep these alphabetised? `tooltip` should go after `order` here.

File: connectors/sources/jira.py [hunk 2]
```diff
@@ -269,6 +269,7 @@ def get_default_configuration(cls):
             "projects": {
                 "display": "textarea",
                 "label": "Jira project keys",
+                "tooltip": "This configurable field is ignored when Advanced Sync Rules are used.",
```
Comment (navarone-feekery) [inline connectors/sources/jira.py:272]: Location here is fine but I notice there's no `order` for `projects`. Could you also add one while you're doing this PR?

**Review by navarone-feekery — CHANGES_REQUESTED** [review]

File: connectors/sources/jira.py (hunk 2 above)
Comment (praveen-elastic) [reply connectors/sources/jira.py:273]: Updated with commit id: [c44df23](https://github.com/elastic/connectors-python/pull/1025/commits/c44df23c9c71ee41d671df666506142700f46f96)

File: connectors/sources/confluence.py (hunk 1 above)
Comment (praveen-elastic) [reply connectors/sources/confluence.py]: Updated with commit id: [803c757](https://github.com/elastic/connectors-python/pull/1025/commits/803c7574c4f2fd2ddce1c94e9867accbbb87edd2)

**Review by navarone-feekery — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**